tests/parser/test_parser.py

```diff
@@ -627,14 +628,15 @@ def test_tsv_doc_preprocessor(caplog):
 
     # Test tsv document
     docs_path = "tests/data/various_format/tsv_format.tsv"
-    preprocessor = TSVDocPreprocessor(docs_path, max_docs=1)
```

Comment:
We should still have a test which explores the code path of max_docs actually limiting the number of documents parsed.